src/utils.js

```diff
@@ -58,6 +58,33 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  variables.forEach((variable) => {
+    if (variable.name === node.name) {
+      scopeVar = variable;
+    }
+  });
+
+  if (typeof scopeVar === 'undefined' ||
+      typeof scopeVar.defs === 'undefined' ||
+      typeof scopeVar.defs[0] === 'undefined' ||
+      typeof scopeVar.defs[0].node.init === 'undefined' ||
```

Comment:
Hard to say. I haven't seen an AST node with a defs field that didn't have a node as part of one of those objects but it's been difficult to nail down the schema of any of this. I can throw that check in there just to be sure. This is already looking pretty nasty though, is there a better way of doing this?